src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -1730,18 +1730,39 @@ static public String extractEmailFromToHeader(String header) {
 		 */
 
 		protected JSONObject createGoogleGroupForSite(String sessionId, String siteId) {
-			String server = env.getProperty(Utils.GGB_SERVER_NAME);
+
+			GGBApiWrapper ggb = establishGGBConnection();
 
 			JSONObject googleGroupSettings = getGoogleGroupSettings(sessionId, siteId);
 
-			GGBApiWrapper ggb = new GGBApiWrapper(server,null);
-			String url = String.format("/groups/%s",googleGroupSettings.getString("email"));
-			HttpResponse ggb_response = ggb.put_request(url, googleGroupSettings.toString());
-			// TODO: error checking.
+			String new_group_url = String.format("/groups/%s",googleGroupSettings.getString("email"));
+			ApiResultWrapper arw = ggb.put_request(new_group_url,googleGroupSettings.toString());
+
 			log.warn("check for errors");
-			log.debug("group add ggb_response: {} ",ggb_response.toString());
 
-			return googleGroupSettings;
+			if (arw.getStatus() == SC_OK) {
+				return googleGroupSettings;
+			}
+			return null;
+		}
+
+		public GGBApiWrapper establishGGBConnection() {
+			
+			String server = env.getProperty(Utils.GGB_SERVER_NAME);
+			String ggb_authinfo_username = env.getProperty(Utils.GGB_AUTHINFO_BASICAUTH_USERNAME);
```

Comment:
Good question.  We all need to agree on what to do when there is an error that doesn't make CPM useless but does mean there is no point in continuing the current task.  E.g.
can't get to microservice, but resource migration is fine.